tests/test_blank_footprint.py

```diff
@@ -0,0 +1,218 @@
+            self._facility_id = facility_id
+            self.blank = blank_footprint
+            if self.blank:
+                self.EXPECTED_APPT_FOOTPRINT = self.make_appt_footprint({})
+            else:
+                self.EXPECTED_APPT_FOOTPRINT = self.make_appt_footprint({'ConWithDCSA': 1})
+
+
+
+        def apply(self, person_id, squeeze_factor):
+            if squeeze_factor != np.inf:
+                # Check that this appointment is being run and run not with a squeeze_factor that signifies that a cadre
+                # is not at all available.
+                self.did_run = True
+
+
+    hsi_dummy = HSI_Dummy_no_blank(module=module, person_id=0)
+    hsi_dummy.initialise()
+    hsi_dummy.facility_info = module.sim.modules['HealthSystem']._facility_by_facility_id[facility_id]
+    return hsi_dummy
+
+
+def get_dummy_hsi_event_instance_blank_footprint(module, facility_id=None):
+    """Make an HSI Event that runs for person_id=0 in a particular facility_id and requests consumables,
+    and for which its parent is the identified module."""
+
+    _facility_level = find_level_of_facility_id(facility_id)
+
+    class HSI_Dummy_with_blank(HSI_Event, IndividualScopeEventMixin):
+        def __init__(self, module, person_id):
+            super().__init__(module, person_id=person_id)
+            self.TREATMENT_ID = 'Dummy'
+            self.ACCEPTED_FACILITY_LEVEL = _facility_level
+            self.EXPECTED_APPT_FOOTPRINT = self.make_blank_appt_footprint()
+            self._facility_id = facility_id
+
+        def apply(self, person_id, squeeze_factor):
+            if squeeze_factor != np.inf:
+                # Check that this appointment is being run and run not with a squeeze_factor that signifies that a cadre
+                # is not at all available.
+                self.did_run = True
+        def make_blank_appt_footprint(self):
+            return {}
+
+    hsi_dummy = HSI_Dummy_with_blank(module=module, person_id=0)
+    hsi_dummy.initialise()
+    hsi_dummy.facility_info = module.sim.modules['HealthSystem']._facility_by_facility_id[facility_id]
+    return hsi_dummy
+
+
+def test_outputs_to_log_no_blank(tmpdir):
+    """Check that logging from Consumables is as expected."""
+    intrinsic_availability = {0: 1.0, 1: 0.0}
+
+    sim = get_sim_with_dummy_module_registered(
+        data=create_dummy_data_for_cons_availability(
+            intrinsic_availability=intrinsic_availability,
+            months=[1],
+            facility_ids=[0]),
+        tmpdir=tmpdir,
+        run=False
+    )
+
+    # Edit the `initialise_simulation` method of DummyModule so that, during the simulation, an HSI is run.
+    def schedule_hsi(sim):
+        """Drop-in replacement for `initialise_simulation` in the DummyModule module."""
+        # Make the district for person_id=0 such that the HSI will be served by facility_id=0
+        sim.population.props.at[0, 'district_of_residence'] = mfl.set_index('Facility_ID').loc[0].District
```

Comment:
This line has also been repeated on line 200 in the function called `test_outputs_to_log_blank()`. I would suggest that we bundle this repetition in one function as seen in the comment above and then just call it when needed. We can simply call it as follows.

 ```suggestion
        sim = set_person_district_id(sim)
```